.gitignore

```diff
@@ -7,4 +7,7 @@ src/*.dll
 GPATH
 GRTAGS
 GTAGS
-*~
\ No newline at end of file
+*~
+config.log
+config.status
+src/Makevars
```

Comment:
These three transient files are removed by `./cleanup` which is why we survived 10+ years without them here.  

Most people I work with naturally keep PRs minimal.  Here I also happen to think it is better not to have them listed as it makes it clearer that the repo is _not_ clean'ed when one sees them.

It's minor but I like it that way so I restored it.